Documentation/EtcdWatch.md

```diff
@@ -0,0 +1,110 @@
+# Overview
+
+The EtcdWatch provide methods to watch on a key and cancel a watcher. If the watcher is disconnected on error, it will be resumed automatically.
+
+# Goals
+
+1. The watch client should process watch request, the watch client create watch request for client with key, option(revision, delete only, put only and end key) and register callback for watch request.
+
+2. Notification, when watch client receive event from etcd server, it will should call registered callback.
+
+3. Resume, when the watch client disconnect with etcd server, the etcd client should automatically resume all the watch requests and get notification from the latest received revision.
```

Comment:
@heyitsanthony Sorry, I use uncorrect word, it should be ' get events from the latest received revision', set the new watch option with revision = last_revision + 1.